### ISSUE TITLE
Markdown parser when having link lists as TOC

### ISSUE THREAD

**Issue by flobee** [go-gitea/gitea#7629]
- Can you reproduce the bug at https://try.gitea.io:
https://try.gitea.io/company/markdown-checks/src/branch/master/README.md

Hello all

I found out that gitea not always renders markdown (e.g. README.md file) as expected for internal link/ankers when using a table of contents generator. 
It works and it doesnt. 

Maybe a restriction in length or incomplete parser? See link above. 

The only difference in my examples: The one that dont work has only one root element.
The working example has mor than one. Dropping a first level solves it but i think this is unusual or am i wrong?

Kind regards florian

**Comment by silverwind** [comment]
Your example is 404.

**Comment by flobee** [comment]
Try this [1], i updated the link an hour ago because it was the edit link
[1] https://try.gitea.io/company/markdown-checks/src/branch/master/README.md

**Comment by flobee** [comment]
> Your example is 404.

Try this [1], i updated the link an hour ago because it was the edit link
[1] https://try.gitea.io/company/markdown-checks/src/branch/master/README.md

Now it works: it was private! sorry!

**Comment by mrsdizzie** [comment]
It is a little hard to follow the description but I think this is maybe the upstream issue:

 https://github.com/russross/blackfriday/issues/329

**Comment by flobee** [comment]
@mrsdizzie Yes, you are right! I could try to send the number space chars i set to make it clear! Sry!!! i' updated my rules for 3 chars indent using the `doctoc` tool, which currently works.. in other situations things dont work. I try to make the report better next time. 

Kind regards Florian
